### ISSUE TITLE
Write Web Scraping Script to Harvest Existing Gallery2 Site

### ISSUE THREAD

**Issue by truephredd** [plagmada/plagmada-archives#23]
Using Python & Beautiful Soup, create a script that crawls the existing website to gather specified metadata and image data.

**Comment by truephredd** [comment]
Current status:  I have logic that spans the gallery recursively, reaching each album page.
Next up is to create a function that will collect the images and metadata associated with them.

**Comment by truephredd** [comment]
Current status:  My code spans the entire archive correctly now.  There'd been an issue that I needed to clean up, but it does it now (although the connection to the site isn't reliable enough that I've made it to the end yet).  

I've also checked out what format is optimal for import into both Omeka and Collective Access, and the answer is the same:  XML.  So, now I just need to start adding the code that'll do that.

@kwhite2 @riverfr0zen @timhutchings

**Comment by timhutchings** [comment]
this sounds exciting!

**Comment by truephredd** [comment]
Well, it's progress.

**Comment by truephredd** [comment]
Checking in today just to keep my productivity tracker happy.  It wants 5 minutes.  Planning on using tomorrow to do some coding and put in real time.

Currently, as per the slack, I've gotten my script to the point that it can scan through the entire current archive successfully, deal with bad links and print out metadata to the screen.  My next step is to start saving images locally and then to take the metadata and save it in a format that can be uploaded by both our target applications (XML).  

I've also laid out what's going on in the script in pseudocode (link: https://plagmada.slack.com/files/phredd/F3PRKJLJF/pseudocode.jpg), which also describes what data and metadata I'm targeting.

**Comment by kwhite2** [comment]
Excuse the libsplaining if you remember this but just didn't mention it for the sake of brevity:


Before uploading, there is likely going to need to be serious reformatting of the XML according to the Dublin Core schema, since most of the relevant metadata for the digital objects is contained in "notes" fields. It's going to have to be separated out and re-formatted to work well with the metadata fields in the new content management system.


Since there's no way on this green earth I'd be able to spend the time doing this on my own at the moment, I have a thought. My suggestion is that, once we're ready and the metadata is harvested from the old system, I'll sit down, take a look at it, and then we do an edit-a-thon. I'll train everyone and we can all set aside a weekend to sit down and blaze through crosswalking and reformatting the information to ingest into the new system.


Krista

________________________________

**Comment by truephredd** [comment]
Yep!  I knew I'd need help with the eventual massaging of the metadata.
Sounds like I should not worry about that for now and just get things going
to produce a package with our data and metadata in a format we can all look
at.

I can always update the script and take another pass once we have a better
idea of how to format things.  I like the edit-a-thon proposal!

Cheers!

**Comment by timhutchings** [comment]
Neat!

**Comment by kwhite2** [comment]
Yep, just keep on with the script writing. We can worry about metadata once we have some preliminary XML to look at.

Thanks for your work on this Phredd!

Krista
